### ISSUE TITLE
[Bug] The mismatched labels configured in RayCluster Helm chart

### ISSUE THREAD

**Issue by thien-nguyenthanh** [ray-project/kuberay#495]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/kuberay/issues) and found no similar issues.

### KubeRay Component

ray-operator

### What happened + What you expected to happen

I am using Kuberay helm chart to create a Ray cluster. 
The service created by Ray operator is mislabeled so Ray worker cannot connect to Ray Head.

**Ray head's labels:**
```
apiVersion: v1
kind: Pod
metadata:
  annotations:
    ray.io/ft-enabled: "false"
    ray.io/health-state: ""
  creationTimestamp: "2022-08-19T07:28:07Z"
  generateName: ray-cluster-head-
  labels:
    app.kubernetes.io/created-by: kuberay-operator
    app.kubernetes.io/instance: ray-cluster
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: ray
    app.kubernetes.io/version: "1.0"
    groupName: headgroup
    helm.sh/chart: ray-cluster-0.3.0
```

**Service's labels**
```
apiVersion: v1
kind: Service
spec:
  clusterIP: 10.0.53.138
  ...
  selector:
    app.kubernetes.io/created-by: kuberay-operator
    app.kubernetes.io/name: kuberay
    ray.io/cluster: ray-cluster
    ray.io/identifier: ray-cluster-head
    ray.io/node-type: head
```

Ray worker logs:
`2022-08-19 00:29:11,420	WARNING utils.py:1243 -- Unable to connect to GCS at ray-cluster-head-svc:6379. Check that (1) Ray GCS with matching version started successfully at the specified address, and (2) there is no firewall setting preventing access.`

**=> Ray worker cannot start because it is not connected to Ray head service**

### Reproduction script

- AKS 1.23
- Both Kuberay v0.3.0 & v0.3.0-rc.2

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

**Comment by Jeffwan** [comment]
@thien-nguyenthanh Just like to confirm, seems you are using helm chart to create the ray-cluster, right?

**Comment by thien-nguyenthanh** [comment]
> 

That's right.

**Comment by jeethridge** [comment]
I've been experiencing the same issue. I think this is definitely a helm chart issue. I believe I've found a workaround whereby you can set the following in your local `values.yaml` file:

```yaml
nameOverride: "kuberay"
...
```

The default name is generated in :
`./helm-chart/ray-cluster/templates/_helpers.tpl` 
fby splitting the Chart.Name value with the `-` delimiter and taking the first part of the name, which happens to be `ray`. 

One possible fix is to update the corresponding Chart.yaml such that the name field is `kuberay-cluster` instead of `ray-cluster`

i.e. in `./helm-chart/ray-cluster/Chart.yaml`

```yaml
apiVersion: v1
appVersion: "1.0"
description: A Helm chart for Kubernetes
name: kuberay-cluster
version: 0.3.0
icon: https://github.com/ray-project/ray/raw/master/doc/source/images/ray_header_logo.png
```

Would that be an acceptable solution going forward?

**Comment by DmitriGekhtman** [comment]
This was fixed in #530!
